reactor-netty-core/src/main/java/reactor/netty/channel/FluxReceive.java

```diff
@@ -182,11 +183,19 @@ final void startReceiver(CoreSubscriber<? super Object> s) {
 			}
 			else {
 				if (log.isDebugEnabled()) {
-					log.debug(format(channel, "{}: Only one connection receive subscriber allowed."), this);
+					log.debug(format(channel, "{}: Ignoring additional inbound receiver subscriber."), this);
 				}
-				Operators.error(s,
-						new IllegalStateException(
-								"Only one connection receive subscriber allowed."));
+
+				StringBuilder msg = new StringBuilder("Ignoring additional inbound receiver subscriber.");
+				msg.append(" Current flux state=[");
+				msg.append("terminated=").append(inboundDone);
+				msg.append(",cancelled=").append(isCancelled());
+				msg.append(",pending=").append(pending);
+				msg.append(",error=").append(inboundError != null);
+				msg.append("].");
```

Comment:
not exactly, the `FluxReceive.toString()` method is displaying the inboundError Message (see `inboundError=" + inboundError`, so the inboundError.toString() method is displayed):

```
	public String toString() {
		return "FluxReceive{" +
				"pending=" + getPending() +
				", cancelled=" + isCancelled() +
				", inboundDone=" + inboundDone +
				", inboundError=" + inboundError +
				'}';
	}
```
so if we reuse the FluxReceive.toString() method when constructing the IllegalStateException, we would end up with something like this (here inboundError is an IOException for example), which is probably a bit too much verbose:

```
java.lang.IllegalStateException: Ignoring additional inbound receiver subscriber. Current flux state=FluxReceive{pending=1, cancelled=false, inboundDone=true, inboundError=java.io.IOException: Inbound Exception}
	at reactor.netty.channel.FluxReceive.startReceiver(FluxReceive.java:197)
	at reactor.netty.channel.FluxReceive.subscribe(FluxReceive.java:148)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:62)
...
Caused by: java.io.IOException: Inbound Exception
	at reactor.netty.channel.FluxReceiveTest.testFluxReceive(FluxReceiveTest.java:188)
	... 86 more
```

so, here, what I mean is that the `inboundError=java.io.IOException: Inbound Exception` is redondant (because it is also displayed in the root cause exception at the bottom of the stacktrace. That's why I'm formatting a bit differently, and if an error is present, only `error=true` will be displayed in the message header, and the real error will be displayed at the bottom of the stacktrace:

```
java.lang.IllegalStateException: Ignoring additional inbound receiver subscriber. Current flux state=[terminated=true,cancelled=false,pending=1,error=true].
	at reactor.netty.channel.FluxReceive.startReceiver(FluxReceive.java:197)
	at reactor.netty.channel.FluxReceive.subscribe(FluxReceive.java:148)
	at reactor.core.publisher.InternalFluxOperator.subscribe(InternalFluxOperator.java:62)
...
Caused by: java.io.IOException: Inbound Exception
	at reactor.netty.channel.FluxReceiveTest.testFluxReceive(FluxReceiveTest.java:188)
	... 86 more
```

WDYT ?

Reply:
forget about my previous comments, you are right and we need to centralize the formatting in one single toString method. 

Please check [3633de8](https://github.com/reactor/reactor-netty/pull/2585/commits/3633de88ef374fd832c6209a6806bfa5f53dece8)